includes/class-donations.php

```diff
@@ -379,5 +380,122 @@ public static function process_donation_form() {
 		\wp_safe_redirect( \wc_get_page_permalink( 'checkout' ) );
 		exit;
 	}
+
+	/**
+	 * Create the donation page prepopulated with CTAs for the subscriptions.
+	 *
+	 * @return int Post ID of page.
+	 */
+	public static function create_donation_page() {
+		$revenue_model = 'donations';
+
+		$intro           = esc_html__( 'With the support of readers like you, we provide thoughtfully researched articles for a more informed and connected community. This is your chance to support credible, community-based, public-service journalism. Please join us!', 'newspack' );
+		$content_heading = esc_html__( 'Donation', 'newspack' );
+		$content         = esc_html__( "Edit and add to this content to tell your publication's story and explain the benefits of becoming a member. This is a good place to mention any special member privileges, let people know that donations are tax-deductible, or provide any legal information.", 'newspack' );
+
+		$intro_block           = '
+			<!-- wp:paragraph -->
+				<p>%s</p>
+			<!-- /wp:paragraph -->';
+		$content_heading_block = '
+			<!-- wp:heading -->
+				<h2>%s</h2>
+			<!-- /wp:heading -->';
+		$content_block         = '
+			<!-- wp:paragraph -->
+				<p>%s</p>
+			<!-- /wp:paragraph -->';
+
+		$page_content = sprintf( $intro_block, $intro );
+		if ( 'donations' === $revenue_model ) {
+			$page_content .= self::get_donations_block();
+		} elseif ( 'subscriptions' === $revenue_model ) {
+			$page_content .= self::get_subscriptions_block();
+		}
+		$page_content .= sprintf( $content_heading_block, $content_heading );
+		$page_content .= sprintf( $content_block, $content );
+
+		$page_args = [
+			'post_type'      => 'page',
+			'post_title'     => __( 'Support our publication', 'newspack' ),
+			'post_content'   => $page_content,
+			'post_excerpt'   => __( 'Support quality journalism by joining us today!', 'newspack' ),
+			'post_status'    => 'draft',
+			'comment_status' => 'closed',
+			'ping_status'    => 'closed',
+		];
+		return wp_insert_post( $page_args );
```

Comment:
Currently this creates a page every time the wizard is loaded; it should set the donation page after creating it.

```suggestion
		$page_id = wp_insert_post( $page_args );
		if ( is_numeric( $page_id ) ) {
			self::set_donation_page( $page_id );
		}
		return $page_id;
```

Reply:
Great catch. https://github.com/Automattic/newspack-plugin/pull/228/commits/23e95964e09ad03f9a5cda8b6a617da329944843